packages/wallet-sdk/src/sign/scw/SCWSigner.ts

```diff
@@ -78,30 +84,99 @@ export class SCWSigner implements Signer {
+
+    if (!config?.enableAutoSubAccounts) {
+      return;
+    }
+
+    // Get the owner account
+    const { account: owner } = config.toOwnerAccount
+      ? await config.toOwnerAccount()
+      : await getCryptoKeyAccount();
+
+    if (!owner) {
+      throw standardErrors.provider.unauthorized('No owner account found');
+    }
+
+    // Set the capabilities for the sub account
+    const capabilities = {
+      addSubAccount: {
+        account: {
+          type: 'create',
+          keys: [
+            {
+              type: owner.address ? 'address' : 'webauthn-p256',
+              key: owner.address || owner.publicKey,
+            },
+          ],
+        },
+      },
+    };
+
+    // Store the owner account and capabilities in the non-persisted config
+    store.subAccountsConfig.set({
+      capabilities,
+    });
+  }
+
+  // TODO: Properly type the return value
+  async request(request: RequestArguments): Promise<any> {
     if (this.accounts.length === 0) {
       switch (request.method) {
         case 'wallet_switchEthereumChain': {
           assertParamsChainId(request.params);
           this.chain.id = Number(request.params[0].chainId);
           return;
         }
-        case 'wallet_connect':
-        case 'wallet_sendCalls':
+        case 'wallet_connect': {
+          // Wait for the popup to be loaded before making async calls
+          await this.communicator.waitForPopupLoaded?.();
+          await this.initSubAccountConfig();
+          const modifiedRequest = injectRequestCapabilities(
+            request,
+            store.subAccountsConfig.get()?.capabilities ?? {}
+          );
+          return this.sendRequestToPopup(modifiedRequest);
+        }
+        case 'wallet_sendCalls': {
           return this.sendRequestToPopup(request);
-        default:
-          throw standardErrors.provider.unauthorized();
+        }
       }
     }
 
     if (this.shouldRequestUseSubAccountSigner(request)) {
-      return this.sendRequestToSubAccountSignerSigner(request);
+      return this.sendRequestToSubAccountSigner(request);
     }
 
     switch (request.method) {
-      case 'eth_requestAccounts':
+      case 'eth_requestAccounts': {
+        const subAccountsConfig = store.subAccountsConfig.get();
+        if (subAccountsConfig?.enableAutoSubAccounts) {
+          // Wait for the popup to be loaded before making async calls
+          await this.communicator.waitForPopupLoaded?.();
+          await this.initSubAccountConfig();
+
+          // This will populate the store with the sub account
+          await this.request({
+            method: 'wallet_connect',
+            params: [
+              {
+                version: 1,
+                capabilities: subAccountsConfig?.capabilities ?? {},
+              },
+            ],
+          });
+
+          const subAccount = store.subAccounts.get();
+          assertPresence(subAccount, standardErrors.provider.unauthorized('No sub account found'));
```

Comment:
this block is called when: dapp calling `eth_requestAccounts` even tho SDK already connected and have cache, usually this edge case happens when dapp's storage gets cleaned somehow.

i wonder if we should move this logic to handshake function in SCWSigner and switch to check if there's already sub account cache in SDK here

Reply:
good catch. i realized we're not making use of the cache here yet but i am worried about what happens if a user wants to switch global accounts, which should result in a new sub account being created - how do they clear this cache?

Reply:
we currently don't support multi global account so we should be fine? when we do support it, i think each switch should clear all caches beforehand?